java-client/src/main/java/org/opensearch/client/transport/httpclient5/ApacheHttpClient5Transport.java

```diff
@@ -1013,4 +1020,72 @@ public InputStream asInput() {
             return new ByteArrayInputStream(this.buf, 0, this.count);
         }
     }
+
+    /**
+     * Wrap the exception so the caller's signature shows up in the stack trace, taking care to copy the original type and message
+     * where possible so async and sync code don't have to check different exceptions.
+     */
+    private static Exception extractAndWrapCause(Exception exception) {
```

Comment:
@andrewparmet I am not very proud of this code, but it seems to be doing the job for now, regarding AwsSdk2Transport, I am not very familiar with it (@dblock is) but we could split this change in tow minimize any potential risks.